src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -161,10 +167,11 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
      * Parses block comment content as javadoc comment.
      * @param blockComment
      *        block comment content.
+     * @param result Reference to object returned by {@link #parseJavadocAsDetailNode(DetailAST)}
      * @return parse tree
      * @noinspection deprecation
      */
-    private ParseTree parseJavadocAsParseTree(String blockComment) {
+    private ParseTree parseJavadocAsParseTree(String blockComment, ParseStatus result) {
```

Comment:
> Change parseJavadocAsParseTree to createJavadocParser and ...

Move `firstNonTightHtmlTag` to a new method `getFirstNonTightHtmlTag(parser)` and do all other work inside `parseJavadocAsDetailNode`.
I like this the best.